src/teacher/StudentReadingInsights.js

```diff
@@ -34,28 +50,17 @@ export default function StudentReadingInsights({ api }) {
     // eslint-disable-next-line
   }, [forceUpdate]);
 
-  useEffect(() => {
-    api.getCohortsInfo((cohortInfo) => {
-      let currentCohort = cohortInfo.filter((each) => each.id === cohortID);
-      setCohortLang(currentCohort[0].language_name);
-    });
-    api.getStudentInfo(
-      studentID,
-      cohortID,
-      selectedTimePeriod,
-      (studentInfo) => setStudentName(studentInfo.name),
-      (error) => console.log(error)
-    );
-    // eslint-disable-next-line
-  }, []);
-
   const customText =
     readArticles &&
     studentName +
       strings.studentHasRead +
       articleCount +
       strings.textsInTheLastPeriod;
 
+  if (studentName === "" || articleCount === -1) {
```

Comment:
would it make sense to initialize both these with null (i.e. `useState()`)? especially the -1 is not much more meaningful than `null` I think.

Reply:
True. Fixed in next pr